cpp/src/arrow/io/file.h

```diff
@@ -183,10 +183,12 @@ class ARROW_EXPORT MemoryMappedFile : public ReadWriteFileInterface {
 
   /// Create new file with indicated size, return in read/write mode
   static Status Create(const std::string& path, int64_t size,
-                       std::shared_ptr<MemoryMappedFile>* out);
+                       std::shared_ptr<MemoryMappedFile>* out,
+		       const off_t offset = 0);
```

Comment:
Adding `offset` to this function is a little bit odd. Is there an immediate use case for this (I would think that mapping a portion of a pre-existing file would be the main thing)